modules/platforms/dotnet/Apache.Ignite/Internal/Table/Table.cs

```diff
@@ -67,106 +67,377 @@ public Table(string name, Guid id, ClientFailoverSocket socket)
         public Guid Id { get; }
 
         /// <inheritdoc/>
-        public async Task<IIgniteTuple?> GetAsync(IIgniteTuple keyRec)
+        public async Task<IIgniteTuple?> GetAsync(IIgniteTuple key)
         {
-            IgniteArgumentCheck.NotNull(keyRec, nameof(keyRec));
+            IgniteArgumentCheck.NotNull(key, nameof(key));
 
             var schema = await GetLatestSchemaAsync().ConfigureAwait(false);
 
             using var writer = new PooledArrayBufferWriter();
-            Write(writer.GetMessageWriter());
+            WriteTuple(writer, schema, key, keyOnly: true);
 
             using var resBuf = await _socket.DoOutInOpAsync(ClientOp.TupleGet, writer).ConfigureAwait(false);
-            return Read(resBuf.GetReader());
+            var resSchema = await ReadSchemaAsync(resBuf, schema).ConfigureAwait(false);
 
-            void Write(MessagePackWriter w)
+            return ReadValueTuple(resBuf, resSchema, key);
+        }
+
+        /// <inheritdoc/>
+        public async Task<IList<IIgniteTuple>> GetAllAsync(IEnumerable<IIgniteTuple> keys)
+        {
+            IgniteArgumentCheck.NotNull(keys, nameof(keys));
+
+            var schema = await GetLatestSchemaAsync().ConfigureAwait(false);
+
+            using var writer = new PooledArrayBufferWriter();
+            var count = WriteTuples(writer, schema, keys, keyOnly: true);
+
+            if (count == 0)
```

Comment:
Should not we check this part earlier? Like, we can return empty collection without requesting schema if the `keys` is empty.

Reply:
It is a bit cumbersome because `IEnumerable` does not have `Count`, but I've fixed it.